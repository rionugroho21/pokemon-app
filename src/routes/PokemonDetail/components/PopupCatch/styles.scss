.pokemon-naming {
  margin-top: 10px;
  border: 1px solid #aeaeae;
  border-radius: 4px;
  width: 300px;
  height: 32px;
  padding: 0px 8px;

  &.error{
    border: 1px solid #ff1c1c;
  }
}

.pokemon-error {
  font-size: 14px;
  margin-top: 6px;
  color: #ff1c1c;
}

.btn-save {
  width: 120px;
  padding: 12px 0px;
  border: none;
  background-color: #007bff;
  border-radius: 5px;
  color: #fff;
  text-align: center;
}

.btn-close {
  width: 120px;
  padding: 12px 0px;
  border: none;
  background-color: #6c757d;
  border-radius: 5px;
  color: #fff;
  text-align: center;
}