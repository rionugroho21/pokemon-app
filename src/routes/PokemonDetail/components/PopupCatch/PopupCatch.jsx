import React from 'react';
import Modal from 'react-bootstrap/Modal';

import './styles.scss';

export const PopupCatch = props => {
  const {
    isCaught,
    message,
    pokemonDetail,
    isExist,
    isNullName,
    handleModal,
    handleSave,
    handleNaming
  } = props;

  return (
    <Modal
      show={isCaught}
      onHide={() => handleModal(false)}
      size="lg"
      aria-labelledby="contained-modal-title-vcenter"
      centered
    >
      <Modal.Header closeButton>
        <Modal.Title id="contained-modal-title-vcenter">
          {message}
        </Modal.Title>
      </Modal.Header>
      <Modal.Body>
        <h4>Make a name for your pokemon</h4>
        <input
          className={`pokemon-naming ${(isExist || isNullName) && 'error'}`}
          type='text'
          onChange={e => handleNaming(e)}
        />
        {isExist && <p className="pokemon-error">The name already exists</p>}
        {isNullName && <p className="pokemon-error">Name cannot be empty</p>}
      </Modal.Body>
      <Modal.Footer>
        <button className="btn-save" onClick={() => handleSave(pokemonDetail)}>Save</button>
        <button className="btn-close" onClick={() => handleModal(false)}>Close</button>
      </Modal.Footer>
    </Modal>
  );
};

export default PopupCatch;