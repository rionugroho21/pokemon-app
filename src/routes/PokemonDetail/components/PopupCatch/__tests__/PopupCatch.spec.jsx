import React from 'react';
import { render, cleanup } from '@testing-library/react';

import PopupCatch from '../PopupCatch';

const props = {
  isCaught: false,
  message: 'Failed to catch the Pokemon',
  pokemonDetail: {},
  isExist: false,
  handleModal: jest.fn(),
  handleSave: jest.fn(),
  handleNaming: jest.fn()
};

describe('PopupCatch Page', () => {
  afterEach(cleanup);

  it('should render page correctly', () => {
    const { asFragment } = render(<PopupCatch {...props} />);
    expect(asFragment()).toMatchSnapshot();
  });
});