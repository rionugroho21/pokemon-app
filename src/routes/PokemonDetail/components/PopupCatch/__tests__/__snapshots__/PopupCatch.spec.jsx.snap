// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PopupCatch Page should render page correctly 1`] = `<DocumentFragment />`;
